src/minesweeperMod/common/Reward.java

```diff
@@ -0,0 +1,55 @@
+package minesweeperMod.common;
+
+import java.util.Random;
+
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import cpw.mods.fml.common.registry.GameRegistry;
+
+public class Reward {
+	private int weight;
+	private RewardItem[] rewardItems;
+	
+	public Reward(String reward) {
+		String[] components = reward.split(";");
+		weight = Integer.parseInt(components[0]);
+		rewardItems = new RewardItem[components.length - 1];
+		for (int i = 0; i < rewardItems.length; i++) {
+			rewardItems[i] = new RewardItem(components[i + 1]);
+		}
+	}
+
+	public ItemStack[] getReward() {
+		ItemStack[] reward = new ItemStack[rewardItems.length];
+		for (int i = 0; i < rewardItems.length; i++) {
+			RewardItem rewardItem = rewardItems[i];
+			Item item = GameRegistry.findItem(rewardItem.modId, rewardItem.itemName);
```

Comment:
This is prone to errors. When the user makes a typo / specifies an invalid item name, the user will only find this out by a crash, and only when this invalid item is picked as a reward. Preferably a sanity check is done in the init phase, with a crash saying which string isn't valid :)